internal/provider/resource_gitlab_project.go

```diff
@@ -1106,41 +1115,43 @@ func resourceGitlabProjectCreate(ctx context.Context, d *schema.ResourceData, me
 		}
 	}
 
-	// If the project is assigned to a group namespace and the group has *default branch protection*
-	// disabled (`default_branch_protection = 0`) then we don't have to wait for one.
-	waitForDefaultBranchProtection, err := expectDefaultBranchProtection(ctx, client, project)
-	if err != nil {
-		return diag.Errorf("Failed to discover if branch protection is enabled by default or not for project %d: %+v", project.ID, err)
-	}
+	if d.Get("skip_wait_for_default_branch_protection").(bool) {
```

Comment:
@timofurrer  - I think you want this to be a "not" check, right? Essentially, you are entering this `if` branch if they _do_ want to wait for the branch protection, which means the `skip` flag should be set to false. It's a weird double negative, but I _think_ that's the intention here.

Reply:
@timofurrer  - Nevermind, ran through a couple examples for myself, I'm good with the logic now.